Hace falta algo que indique con qué entorno vamos a trabajar

Importar lo que hace falta

In [1]:
import pandas as pd

In [2]:
data_url = "../Data/properatti.csv"

In [3]:
data = pd.read_csv(data_url, encoding="utf-8")

In [4]:
data.shape

(121220, 26)

In [5]:
data.isnull().sum()

Unnamed: 0                         0
operation                          0
property_type                      0
place_name                        23
place_with_parent_names            0
country_name                       0
state_name                         0
geonames_id                    18717
lat-lon                        51550
lat                            51550
lon                            51550
price                          20410
currency                       20411
price_aprox_local_currency     20410
price_aprox_usd                20410
surface_total_in_m2            39328
surface_covered_in_m2          19907
price_usd_per_m2               52603
price_per_m2                   33562
floor                         113321
rooms                          73830
expenses                      106958
properati_url                      0
description                        2
title                              0
image_thumbnail                 3112
dtype: int64

Confirmar los valores de la serie 'currency'

La serie price_aprox_usd tiene 20410 null, se eliminan esas observaciones 

In [6]:
data = data[data['price_aprox_usd'].notnull()]

In [7]:
data.shape

(100810, 26)

In [8]:
data.isnull().sum()

Unnamed: 0                        0
operation                         0
property_type                     0
place_name                       23
place_with_parent_names           0
country_name                      0
state_name                        0
geonames_id                   16952
lat-lon                       42262
lat                           42262
lon                           42262
price                             0
currency                          1
price_aprox_local_currency        0
price_aprox_usd                   0
surface_total_in_m2           31944
surface_covered_in_m2         13150
price_usd_per_m2              32193
price_per_m2                  13152
floor                         93894
rooms                         61231
expenses                      87564
properati_url                     0
description                       2
title                             0
image_thumbnail                2250
dtype: int64

In [9]:
data['currency'].value_counts()

USD    87587
ARS    13219
PEN        2
UYU        1
Name: currency, dtype: int64

**Seguimos teniendo observaciones que no son en dólares**

Retenemos sólo las observaciones que correspondan a 'USD'

In [10]:
data = data[data['currency'] == 'USD']

In [11]:
data.shape

(87587, 26)

Determinar si la serie 'price' es igual que 'price_aprox_usd'
La suma de valores True debe ser igual a la cantidad de observaciones

In [12]:
sum(data['price'] == data['price_aprox_usd'])

87587

No hacen falta las series currency, price_aprox_local_currency y price_aprox_usd

In [13]:
data = data.drop(['currency', 'price_aprox_local_currency', 'price_aprox_usd'], axis='columns')

In [14]:
data.shape

(87587, 23)

In [15]:
data.isnull().sum()

Unnamed: 0                     0
operation                      0
property_type                  0
place_name                    23
place_with_parent_names        0
country_name                   0
state_name                     0
geonames_id                16141
lat-lon                    35051
lat                        35051
lon                        35051
price                          0
surface_total_in_m2        27267
surface_covered_in_m2      10355
price_usd_per_m2           27454
price_per_m2               10357
floor                      81278
rooms                      51573
expenses                   74935
properati_url                  0
description                    1
title                          0
image_thumbnail             1542
dtype: int64

Determinar que ocurre con price y surface